clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/config/ArtifactCredentials.java

```diff
@@ -18,17 +18,20 @@
 package com.netflix.spinnaker.clouddriver.artifacts.config;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.ImmutableList;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
 import com.netflix.spinnaker.kork.artifacts.model.Artifact;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.List;
 import java.util.Optional;
 import org.apache.commons.lang3.NotImplementedException;
 
+@NonnullByDefault
 public interface ArtifactCredentials {
   String getName();
 
-  List<String> getTypes();
+  ImmutableList<String> getTypes();
```

Comment:
I'm not sure this needs to be on the interface signature unless there are extra methods on ImmutableList that are being used.

Reply:
There aren't additional methods, but similar to the above comment, it seemed useful to indicate to consumers (and to implementation) that the list returned from `getTypes()` should be immutable.

Reply:
Yeah, generally speaking Google recommends using `Immutable*` as a return type because it gives additional information to people writing clients of this class that the compiler has no way to enforce: namely that if you call any of the mutating methods, you're going to get a runtime exception. Ideally, we'd have a way for the compiler to enforce that, but we don't, so marking the type as immutable is better than nothing. I think that's generally a pretty compelling argument.

Of course internally, there's pretty much no way to compile a binary that doesn't have `com.google.common.collect` linked in, so the additional dependencies aren't a concern.